src/analysis.zig

```diff
@@ -1515,12 +1603,34 @@ fn resolveTypeOfNodeUncached(analyser: *Analyser, node_handle: NodeWithHandle) e
         => {
             const ptr_info = ast.fullPtrType(tree, node).?;
 
+            const sentinel: InternPool.Index = if (ptr_info.ast.sentinel != 0) blk: {
+                // resolveTypeOfNode can also resolve values that returned as indices into the InternPool.
+                const old_resolve_number_literal_values = analyser.resolve_number_literal_values;
+                analyser.resolve_number_literal_values = true;
+                defer analyser.resolve_number_literal_values = old_resolve_number_literal_values;
+                const sentinel = try analyser.resolveTypeOfNode(.{ .node = ptr_info.ast.sentinel, .handle = handle }) orelse break :blk .none;
+                break :blk switch (sentinel.data) {
+                    .ip_index => |payload| payload.index,
+                    else => .none,
+                };
+            } else .none;
```

Comment:
This should be moved into its own function (name suggestion: `resolveInternPoolValue`) because this exact code appears twice in this file.
The `resolveIntegerLiteral` can then call that function.

Reply:
Looking at this again, `resolveIntegerLiteral` and the code above are very similar. I originally moved the logic into a separate `resolveInternPoolValue` function, but rethought things since they only really differ at the return point (`analyser.ip.toInt(payload.index, u64)` vs. `payload.index`)

I tried adding a third argument to `resolveIntegerLiteral` avoid duplicating the common logic instead. Happy to rework it if this isn't the right approach!

Reply:
On second thought, would a wrapper here improve things? 

```diff
diff --git a/src/analysis.zig b/src/analysis.zig
index 69158c7..267bc39 100644
--- a/src/analysis.zig
+++ b/src/analysis.zig
@@ -1198,6 +1198,13 @@ fn resolveIntegerLiteral(analyser: *Analyser, node_handle: NodeWithHandle, kind:
     }
 }

+fn resolveInternPoolValue(analyser: *Analyser, node_handle: NodeWithHandle) !InternPool.Index {
+    if (try resolveIntegerLiteral(analyser, node_handle, .sentinel)) |sentinel| {
+        return @enumFromInt(sentinel);
+    }
+    return .none;
+}
+
 const primitives = std.StaticStringMap(InternPool.Index).initComptime(.{
     .{ "anyerror", .anyerror_type },
     .{ "anyframe", .anyframe_type },
@@ -1613,12 +1620,7 @@ fn resolveTypeOfNodeUncached(analyser: *Analyser, node_handle: NodeWithHandle) e
         => {
             const ptr_info = ast.fullPtrType(tree, node).?;

-            const sentinel: InternPool.Index = blk: {
-                if (try analyser.resolveIntegerLiteral(.{ .node = ptr_info.ast.sentinel, .handle = handle }, .sentinel)) |sent| {
-                    break :blk @enumFromInt(sent);
-                }
-                break :blk .none;
-            };
+            const sentinel = try resolveInternPoolValue(analyser, .{.node = ptr_info.ast.sentinel, .handle = handle});

             const elem_ty = try analyser.resolveTypeOfNodeInternal(.{ .node = ptr_info.ast.child_type, .handle = handle }) orelse return null;
             if (!elem_ty.is_type_val) return null;
@@ -1644,12 +1646,7 @@ fn resolveTypeOfNodeUncached(analyser: *Analyser, node_handle: NodeWithHandle) e

             const elem_count: ?u64 = try analyser.resolveIntegerLiteral(.{ .node = array_info.ast.elem_count, .handle = handle }, .length);

-            const sentinel: InternPool.Index = blk: {
-                if (try analyser.resolveIntegerLiteral(.{ .node = array_info.ast.sentinel, .handle = handle }, .sentinel)) |s| {
-                    break :blk @enumFromInt(s);
-                }
-                break :blk .none;
-            };
+            const sentinel = try resolveInternPoolValue(analyser, .{.node = array_info.ast.sentinel, .handle = handle});

             const elem_ty = try analyser.resolveTypeOfNodeInternal(.{ .node = array_info.ast.elem_type, .handle = handle }) orelse return null;
             if (!elem_ty.is_type_val) return null;
```

Reply:
I would rather expect `resolveIntegerLiteral` to call `resolveInternPoolValue` and not the other way around. I made the change myself and pushed it to your PR.

Reply:
Oh I see what you meant now, thank you!